lib/devise/encryptable/encryptors/pbkdf2.rb

```diff
@@ -1,23 +1,66 @@
-begin
-  module Devise
-    module Encryptable
-      module Encryptors
-        class Pbkdf2 < Base
-          def self.compare(encrypted_password, password, stretches, salt, pepper)
-            value_to_test = self.digest(password, stretches, salt, pepper)
-            Devise.secure_compare(encrypted_password, value_to_test)
-          end
-
-          def self.digest(password, stretches, salt, pepper)
-            hash = OpenSSL::Digest.new('SHA512').new
-            OpenSSL::KDF.pbkdf2_hmac(
-              password.to_s,
-              salt: "#{[salt].pack('H*')}#{pepper}",
-              iterations: stretches,
-              hash: hash,
-              length: hash.digest_length,
-            ).unpack1('H*')
-          end
+module Devise
+  module Encryptable
+    module Encryptors
+      # https://en.wikipedia.org/wiki/PBKDF2
+      # Adapted from https://gitlab.com/gitlab-org/gitlab/-/blob/373f088e755f678478b8dd1627fab908d2641b21/vendor/gems/devise-pbkdf2-encryptable/lib/devise/pbkdf2_encryptable/encryptors/pbkdf2_sha512.rb
+      class Pbkdf2 < Base
```

Comment:
I think my primary discomfort with this is the potential for a name with devise encryptable clash if/when the original PR gets merged. Maybe change the class name to something like `Pbkdf2Hash`?